microsoft/azure-pipelines-tasks
# [NodeToolV0] Update version of "azure-pipelines-tool-lib" to 2.0.3

**Task name**: NodeToolV0

**Description**: Need to update version of `azure-pipelines-tool-lib` to `2.0.3` since it contains fix around downloading Node tool

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

**How it was tested:**
Tested on self-hosted agent by replacing code of task with changes from the PR. Bandwidth connection was limited by application `NetLimiter` to `300 KB/s`. Throttling was simulated by application `TCPView`.
Successful run: https://abtttestorg.visualstudio.com/Other/_build/results?buildId=6276
Run with throttling issues: https://abtttestorg.visualstudio.com/Other/_build/results?buildId=6277

**Test Cases:**
Test pipeline
```yaml
strategy:
  matrix:
    'Node16':
      SomeVariable: 'somevalue'
    'Node10':
      AGENT_USE_NODE10: 'true'

steps:
- task: NodeTool@0
  inputs:
    versionSource: 'spec'
    versionSpec: '16.10.x'
```
1. Start self-hosted agent
2. Replace task `NodeTool` with changes from the PR
3. Limit bandwidth connection to `300 KB/s` (`NetLimiter`, `@sitespeed.io/throttle`, etc.)
4. Run pipeline on the agent
5. During execution of task `NodeTool` close TCP connection to `nodejs.org` server (`TCPView`, `tcpkill`, etc.)

Expected Result:
Task `NodeTool` must be failed with error `Aborted`
